client.go

```diff
@@ -422,7 +442,59 @@ func (c *Client) Rcpt(to string, opts *RcptOptions) error {
 	if err := validateLine(to); err != nil {
 		return err
 	}
-	if _, _, err := c.cmd(25, "RCPT TO:<%s>", to); err != nil {
+	cmdStr := "RCPT TO:<%s>"
+	params := []interface{}{to}
+	if _, ok := c.ext["DSN"]; ok && opts != nil {
+		if opts.Notify != nil && len(opts.Notify) != 0 {
+			val := " NOTIFY="
+			seen := map[DSNNotify]struct{}{}
+			for i, v := range opts.Notify {
```

Comment:
Since the server also performs the same checks, maybe worth extracting into a helper function?

Reply:
Done!  I'd like to squash additional commits later.